cmd/copy.go

```diff
@@ -755,6 +770,14 @@ func (cca *cookedCopyCmdArgs) processCopyJobPartOrders() (err error) {
 	// Strip the SAS from the source and destination whenever there is SAS exists in URL.
 	// Note: SAS could exists in source of S2S copy, even if the credential type is OAuth for destination.
 	switch from {
+	case common.ELocation.Local():
+		tmpSrc, err := filepath.Abs(cca.source)
+		if err != nil {
+			return fmt.Errorf("couldn't get absolute path of the source location %s. Failed with errror %s", cca.source, err.Error())
+		}
+
+		jobPartOrder.SourceRoot = cleanLocalPath(trimWildcards(tmpSrc))
```

Comment:
What is trimWildcards here? Does it remove them? I thought our plan was to _fail with an error_ if there were any wildcards in places where we did not expect them.  Otherwise, don't we run the risk that the user thinks we have done one thing, but really we have done another?

Reply:
trimWildcards is intended to trim the wildcards from a local path. We decided to enable wildcards for local usage as that comes effectively for free, but disallow them on non-local sources.

@zezha-msft correct me if I'm wrong.

Reply:
@adreed-msft yes that's what we decided. We only support `/*` for remote sources.

SourceRoot is used to build the actually transfers, we have find it since in the actual job parts, we give relative paths for each transfer.